### ISSUE TITLE
[BUG]: Bug Description

### ISSUE THREAD

**Issue by Stargrazer82301** [PhangsTeam/pjpipe#70]
**Describe the bug**
The newest version of reproject, 0.14.0, is incompatible with PJPipe. Specifically, the `__init__` for `ReprojectArraySubset` now expects `imin`, `imax`, `jmin`, and `jmax` to be provided together as a single variable ,`bounds`.

**To Reproduce**
Any attempt to run the level matching step in PJPipe.

**Expected behavior**
Not crashing.

**Comment by thomaswilliamsastro** [comment]
I think this should be part of a larger 'get everything up to scratch for numpy 2.0' update, which I won't have time for for the next couple of weeks. In the meantime, if you just use the pinned versions of the packages from `pyproject.toml`, things should work for now

**Comment by Stargrazer82301** [comment]
Yep, I just force-downgraded reproject to 0.13.0, and it works fine again. Just wanted to let you know what had happened here, to save you coming across it and digging it out for yourself. 

Honestly, I kindaenjoyed the novelty of finding a crash _not_ directly caused by the numpy 2.0.0 update...
